build.wake

```diff
@@ -0,0 +1,416 @@
+    Nil
+
+  def cmd =
+    def venvDir = relative runDir (freedomESDKVenvDir buildEnv)
+    # Explicitly create the venv with the 'python3.7' binary so that we use
+    # the version of python with pip and venv preinstalled in all build environments.
+    "bash", "-c", "%
+      export FREEDOM_E_SDK_VENV_PATH=$(pwd)/%{venvDir}
+      python3.7 -m venv ${FREEDOM_E_SDK_VENV_PATH}
+      . ${FREEDOM_E_SDK_VENV_PATH}/bin/activate && python3 -m pip install pip==20.0.1 --disable-pip-version-check
+      . ${FREEDOM_E_SDK_VENV_PATH}/bin/activate && python3 -m pip install -r requirements.txt
+      touch ${FREEDOM_E_SDK_VENV_PATH}/.stamp
+      %", Nil
+
+  makePlan cmd inputs
+  | setPlanDirectory runDir
+  | setPlanResources ("python/python/3.7.1", Nil)
+  | setPlanKeep True
+
+# Wake rules which depend upon the Freedom E SDK virtualenv should
+# add `installFreedomESDKVirtualenv` to their Plan inputs.
+export target installFreedomESDKVirtualenv buildEnv =
+  def installedFiles =
+    makeFreedomESDKVirtualenvJob buildEnv
+    | runJob
+    | getJobOutputs
+  FreedomESDKVirtualenv installedFiles
+
+#
+# Freedom E SDK Installation
+#
+
+export tuple FreedomESDK =
+  export Dir:            String
+  export InstalledFiles: List Path
+  export BuildEnv: FreedomESDKBuildEnv
+
+# makeFreedomESDK assembles the components of Freedom E SDK into the directory
+# hierarchy expected by the SDK from their flat layout in the wit workspace.
+# The resulting sdk is located in {destDir}/freedom-e-sdk.
+export def makeFreedomESDK buildEnv destDir =
+  def sdkDir = "{destDir}/freedom-e-sdk"
+  def scriptsDir = "{sdkDir}/scripts"
+
+  def sdkSources =
+    source "freedom-e-sdk/bsp/update-targets.sh",
+    sources "freedom-e-sdk" `.*`
+    | filter (!matches `freedom-e-sdk/(freedom-devicetree-tools|freedom-metal|bsp|software|doc|pip-cache)(/.*)?` _.getPathName)
+    | filter (!matches `freedom-e-sdk/scripts/(elf2hex|devicetree-overlay-generator|ldscript-generator)(/.*)?` _.getPathName)
+    | filter (!matches `freedom-e-sdk/scripts/(cmsis-svd-generator|openocdcfg-generator|esdk-settings-generator)(/.*)?` _.getPathName)
+    | filterWakeAndDotFiles
+
+  def sifiveCryptoLibSources =
+    sources "scl-metal" `.*`
+    | filterWakeAndDotFiles
+
+  def freeRTOSSources =
+    sources "FreeRTOS-metal" `.*`
+    | filterWakeAndDotFiles
+
+  def systemViewSources =
+    sources "Segger_SystemView-metal" `.*`
+    | filterWakeAndDotFiles
+
+  def elf2hexSources =
+    sources "elf2hex" `.*`
+    | filterWakeAndDotFiles
+
+  def installedFiles = 
+    copyDir "freedom-e-sdk"                   destDir     sdkSources
+    ++ copyDir "scl-metal"                    sdkDir      sifiveCryptoLibSources
+    ++ installFreedomMetal                    sdkDir
+    ++ copyDir "FreeRTOS-metal"               sdkDir      freeRTOSSources
+    ++ copyDir "Segger_SystemView-metal"      sdkDir      systemViewSources
+    ++ copyDir "elf2hex"                      scriptsDir  elf2hexSources
```

Comment:
If these are all coming from different wit/wake repos, then I think we should export a constant in each of these packages that holds the path to the root of the repo (which can be computed from `here`). If some of these repos do not contain any Wake code, then I think @mmjconolly has some code he's been working on in some other repos to attempt to locate git repo with a given name even if that repo contains no Wake code.